zebra/zebra_pbr.c

```diff
@@ -99,6 +99,63 @@ int zebra_pbr_rules_hash_equal(const void *arg1, const void *arg2)
 	return 1;
 }
 
+uint32_t zebra_pbr_ipset_hash_key(void *arg)
+{
+	struct zebra_pbr_ipset *ipset = (struct zebra_pbr_ipset *)arg;
+	uint32_t *pnt = (uint32_t *)ipset->ipset_name;
+
+	return jhash2(pnt, ZEBRA_IPSET_NAME_SIZE, 0x63ab42de);
```

Comment:
jhash2 doc says this:

/* A special optimized version that handles 1 or more of u_int32_ts.
 * The length parameter here is the number of u_int32_ts in the key.
 */
extern u_int32_t jhash2(const u_int32_t *k, u_int32_t length,
			u_int32_t initval);

ZEBRA_IPSET_NAME_SIZE = 32

ipset_name is defined as this:

	char ipset_name[ZEBRA_IPSET_NAME_SIZE];

Shouldn't we pass in the length of 8 instead of 32?

Reply:
yes. you saw that I fixed it later.